config/initializers/ab_tests.rb

```diff
@@ -1,19 +1,46 @@
 # frozen_string_literal: true
 
-require 'ab_test_bucket'
+require 'ab_test'
 
 module AbTests
-  DOC_AUTH_VENDOR = AbTestBucket.new(
+  # Helper method that enables using an Idv::Session to calculate a discriminator value.
+  # If no Idv::Session is available, `nil` is used as the discriminator value.
+  # @returns [Proc]
+  def self.idv_session_discriminator(&block)
+    ->(request:, service_provider:, user:, user_session:) do
+      # If we don't have a logged-in user, we can't use an Idv::Session-based discriminator.
+      return nil if user.blank? || user.is_a?(AnonymousUser)
+
+      return nil if user_session.nil?
+
+      idv_session = Idv::Session.new(
+        current_user: user,
+        service_provider:,
+        user_session:,
+      )
+
+      block.call(idv_session)
```

Comment:
it's marginally faster to use the `yield` keyword and not even declare `&block` in the signature (because then Ruby doesn't to keep a reference on the heap or something)

```suggestion
      yield idv_session
```